### DISCUSSION TITLE
Worklog Report - Several questions for new user

### DISCUSSION THREAD

**Discussion by jeffboon97** [shridhar-tl/jira-assistant#301] [General]
Dear @shridhar-tl  good day to you, I have only recently started using your amazing application to generate useful reports for my management. However I am currently stucked and not able to meet some requirements set by my managements. May need your help to clarify if the following is possible to be achieved?

Question 1 - Is it possible to expand the tickets logged by user by default?
Reason - Because when I export the CSV report, the tickets summary is not shown in the CSV report. This is useful as management would like to know where did the developer logged their time spent.

![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/1a5e9547-fc8c-4850-ba2d-aba55b15dc4b)

Question 2 - Is it possible to show the timeline from the ticket creation date itself?
Reason - As of now, I am only able to filter the date range manually and it is difficult for management to know when was the ticket created

Question 3 - Is it possible to show log time comment column?
Reason - This is important for management to understand why developer spent certain hours or min completing the tasks given.

Question 4 - When export into CSV/Excel, is it possible to grey out holidays or weekend?

Thank you for your time in reading this, appreciate the time spent developing this tool!

**Comment by shridhar-tl** [comment]
Hi @jeffboon97 , Thank you for contacting me and here are my answers:

Question 1 - Is it possible to expand the tickets logged by user by default?
Answer: Yes it is possible now. In Worklog report, Click on show settings icon from top right corner of the page, then switch to "Formatting" tab and then click "Show more options" and you will see a checkbox added at the end. Selecting this would by default expand all the users when you generate a report.
![image](https://github.com/shridhar-tl/jira-assistant/assets/37339683/3e0d64eb-1578-49e4-82b9-f02bb3e58a0a)

Question 2 - Is it possible to show the timeline from the ticket creation date itself?
Answer: Yes is it possible now. Select "Issue Age" field from "Fields list" tab of report configurations and you should have it.
![image](https://github.com/shridhar-tl/jira-assistant/assets/37339683/35cca17b-fe0d-4673-9f14-ff33e1e6decb)

Question 3 - Is it possible to show log time comment column?
Answer: Yes, it is already part of Flat (Groupable) tab of worklog report. But if you are expecting it to be part of "Grouped - [User daywise]" tab then it is not feasible as single row contains worklog for multiple days and so having multiple comments is practically not possible. Any way, as it is available in Flat groupable report, you should be able to export it to excel and share it.

Question 4 - When export into CSV/Excel, is it possible to grey out holidays or weekend?
Answer: CSV is a text file. So it cannot have any formatting. However when it comes to Excel export, I can try it. Will try it and see if I could get something. But it would take couple of more weeks as I am quiet busy at this point for implement this.

Point 1 & 2 is a new implementation and is already available in web version of extension. You can switch to web seamlessly and you should be able to use this feature until it is published to extension in JA v2.49. Once your extension version gets updated to v2.49, you should see it working in extension as well.

**Comment by jeffboon97** [comment]
Good day to you @shridhar-tl ,

I have a few questions after testing out with the worklog report generation, hopefully you can answer.

Q1 - As referring image below, I notice that there's no sorting option available in JA, Therefore when generating the report into Excel. I wasn't able to display the report in chronological order. Wondering is there a way we can sort by worklog created date?
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/a5947b8d-2aca-4128-93ab-9c94b1cc2685)

**Reply by shridhar-tl** [reply]
Hi @jeffboon97 - Good day,
Yes I should be able to do some sorting for the issues.

However, what do you mean by sorting based on worklog creation date. Each story will have have worklog for multiple days and how do you expect the sorting to work based on worklog creation date?

**Reply by jeffboon97** [reply]
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/5b4c97e4-0921-45ab-8c72-94f2f6cf2995)

Basically we are able to see a progressive view of the worklog, sort of like a Gantt Chart.


Thanks for your swift reply as always!

**Reply by shridhar-tl** [reply]
Apologies, unfortunately this image does not fully make sense to me. I am still trying to understand how it would work on different scenarios.

Towards left you have multiple issues. Each row representing single issue. So obviously I can show data of individual stories in single row only.

While that being said, this image means to me that for individual issues worklog is added only on consecutive days. If that is the case should be able to sort it based on worklog date.

But what if, for same issue, worklog is added on 1st day of month, then on 15th of month and then on 28th of month. What if worklog is added randomly on various day of month and this happens on multiple issues? I am not sure if such a case can happen, but just trying to understand.

**Reply by jeffboon97** [reply]
Dear @shridhar-tl , I should have given you some context. The info on the left is actually the worklog author, As we are generating worklog for a single project we know who is the worklog author's role and we want to identify the gaps in between worklog entries. As you can see below, There's a huge gap between 13 to 16 Feb and 20 to 23. 

![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/5ef43817-6fce-4b00-afe2-825c3832b73f)

So my intention was to make the report more pretty in terms of visualization, so we can see the flow from 1 date to another date.
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/34dc73f3-c842-441a-8bb5-b43415fb4b02)

Once we can clear see the gap, then we will drill down to Flat (Groupable) to identify the time logged date, and check with each party why was there a delay in between worklog entries.

I hope you understand my explanation, thank you!

**Reply by jeffboon97** [reply]
Also Dear @shridhar-tl  for your information, I wasn't able to contribute donation to you via PayPal. 

![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/eb8f93bb-672c-414b-a806-156965d0e2e1)

**Reply by shridhar-tl** [reply]
@jeffboon97 - I have done the ordering based on first worklog date for each user and same with tickets as well. Now both user and tickets both would be sorted based on first worklog date. Hope that solves your issue.

This change would be published to extension users as part of JA v2.51. However it is already available in Web for you to try as of now.  If you are an extension user, you can easily switch to Web version and use the latest version of tool always.

**Comment by jeffboon97** [comment]
Good day to you @shridhar-tl,

I am writing to you today to follow up on our recent conversation about the Worklog report. My team is generally satisfied with the report, but we have a few suggestions that we would like to share.

Currently, the Worklog report is sorted by worklog date and categorized into each worklog author
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/ac7b660e-2bad-45e5-9e0b-1ee4a9051601)

With that said we have few suggestions below:
1) We would like to request that the report be sorted by ticket and logged time instead. This would make it easier for us to track the time that each team member is spending on each ticket.

Meaning we are able to know per record basis who logged time and logged under which ticket and the issue type. We can form a Gantt Chart easily without exporting the data.

2) Additionally is it possible to add Issue Type to Report?

**Sample report**
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/31e84714-3c70-45cb-a6a9-3efd062f52e7)

Thank you for your time reading this and I hope you can understand what I am explaining.

Furthermore, please do let me know how can I contribute to you to make this changes possible.

Thank you for your time and consideration.

**Reply by jeffboon97** [reply]
Dear @shridhar-tl  good day to you!

I hope this message finds you well. I am reaching out to understand from you if my enquiries above is feasible? I understand that you are likely busy, your response is much appreciated. 

Thank you!

**Comment by shridhar-tl** [comment]
Hi @jeffboon97 - Have you tried this use case in Web version?

Your request regarding sorting has been implemented 3 weeks back upon your first clarification and I assumed I already responded to this message. But just realized, I did not.

As of now its available only in Web Version. You can have a look at it just by launching the Web version once and see if it works as you expect. If it does work, then I can publish it to extension.

By the way, you can directly use web version without providing any credentials. Web version will work with extension to display the data and it is just like a beta version for extension which contains latest changes always.

**Reply by jeffboon97** [reply]
Dear @shridhar-tl  thank you for your reply, yes I am well aware of the implementation that you did for me from 3 weeks ago. However after using the newer sorted version. There's still some feedback given that we can improve upon it. 

https://github.com/shridhar-tl/jira-assistant/discussions/301#discussioncomment-6110702

Thank you for your time.

**Reply by shridhar-tl** [reply]
Hi @jeffboon97 - Thank you for your response, here is my answer to both your queries,

First to answer your second query regarding issue type: It was intentionally hidden as already issue type icon is displayed along with issue key and summary. So I thought this may not be required. However, it doesn't hurt to still have it and hence added it back. Now you can add it and it should work.

Next, about sorting: I am not sure what is your expectation. Do you say show it without any grouping based on user?

With your screenshot that is what I understand and you are expecting user to be shown as additional column and not as a group. So if two users logged work on same ticket, two rows with same ticket would be displayed with additional column displaying user name.

Please confirm if this is what you are expecting? If yes, this is almost a new report itself and is somewhat a huge effort. I will have to think through it.

**Reply by jeffboon97** [reply]
Dear @shridhar-tl  Good day to you sir,

Regarding your second question, yes that's the intention. We would like the user to be shown as additional column and not grouped, if multiple users recorded time into the same ticket, then there will be duplication. However this duplication will be used to plot into Gantt Chart.

I deeply appreciate your expertise and time to look into implementing it, Please let me know if you have any questions or concerns.


Thank you!

**Reply by shridhar-tl** [reply]
Sure, will think through it. If possible will try to implement it within this week before I publish next version for extension.

**Reply by jeffboon97** [reply]
Good morning @shridhar-tl ,

I hope this message finds you well. I wanted to follow up on the feature we discussed in our previous conversation. I understand that you may be busy, but I wanted to check in and see if you had any updates or questions regarding the feature. 😅

**Reply by shridhar-tl** [reply]
@jeffboon97 - Yes, I had a look at it and started to try customizing existing report to fit in to this need. However facing few difficulties customizing it. I will continue with the implementation this week end as well. Will see if I could get through it by this week end. It doesn't appear to be a straightforward change.

I also came across another existing issue also with the report, while working on this and was looking into that as well. Will try to resolve both by this week end.

**Reply by shridhar-tl** [reply]
I am around 80% done with it. Some minor decisions are pending for completion. Should get it done in next couple of days after which will publish to Web version first. I have added this feature as additional tab in worklog report.

**[ Update ]:**
Not able to get it done as planned due to various issues / difficulties which involved more work than expected.

**Reply by jeffboon97** [reply]
Good day to you @shridhar-tl ! I just wanted to take a moment to express my gratitude and say how super appreciative I am of your effort.

**Reply by jeffboon97** [reply]
Dear @shridhar-tl  , I hope this message finds you well. First, I wanted to thank you for your hard work on the new report. I know you've been working on it pro bono, and I really appreciate your dedication.

I'm writing to ask if you think the report will be ready by this weekend?

If not, please let me know when you think it will be ready.

Thank you again for all your hard work.

**Reply by shridhar-tl** [reply]
@jeffboon97 - This was added as an additional tab named "Flat - [Issue daywise]" under worklog report. I did not get sufficient time last week and so was not able to fully complete it as planned.

However, it does appear to be working as expected for Date range report. Hence I have made it available in Web version.

**What is pending:**
- Currently it is not available when timeframe time is selected as sprint. More work is required to make it available for sprint which I will try to have it implemented in next couple of weeks.
- It is tested with some basic settings and data, however, need more effort to completely test it with various settings like different grouping, formatting and timeframe type.

I would say this is a BETA version of report available in WEB and request you to ensure the data before using it. To use this tab, switch to Web version. Switching to Web is seem less and you need not worry about loosing anything in terms of settings.

If you feel the report is working as expected, you can let me know and I would immediately publish a version for extension users while I continue to work on it for Sprint timeframe type.

**Reply by jeffboon97** [reply]
Dear @shridhar-tl , a great day to you. I have managed to give the report a try. Everything was great, however I did notice an issue maybe it's unique to my business case.


Issue #1 - When I logged once into discussion subtask and if I log again into discussion, although it's different date it will still be shown in the same line.
Expected outcome: Report should create a new line of record instead of added into the same ticket row.
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/6096dbf6-a26c-4180-a3f5-3aaecc24684f)


Suggestion #1 - Can add a column to see the worklog created date so we can see each row in chronological order
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/e4a8619c-e9fb-4ef4-9463-7dd4b3c1438b)


Thank you for your contribution, Do DM me how I can help with donation. Thanks!
![image](https://github.com/shridhar-tl/jira-assistant/assets/124252281/0a350499-d398-4dd8-a1c3-a1ba8709619f)

**Reply by jeffboon97** [reply]
Dear @shridhar-tl good day to you, may I know was you able to look into the issue and suggestion mentioned above? Thanks in advance!

**Reply by shridhar-tl** [reply]
Hi @jeffboon97, Was on a busy week and so far I haven't got a chance to look into it. However I have planned to look into it tomorrow and will try to publish it to web by tomorrow.

**Reply by shridhar-tl** [reply]
Hi @jeffboon97, enable the following configuration under "Formatting" tab and you should see it working.

![image](https://github.com/shridhar-tl/jira-assistant/assets/37339683/b4d3a8a6-4d4d-4f27-9fc3-15d482df4cdd)

Please verify if data is displayed appropriately and let me know, so that I will publish it to extension. I cannot frequently publish the changes to extension as it is a huge effort and so I would like to suggest you to check it well for all your use case. In case of any changes later, it would take more time to publish another version for extension.

**Reply by shridhar-tl** [reply]
Hi @jeffboon97 - Did you get a chance to have a look at the changes?

**Reply by jeffboon97** [reply]
Good day to you @shridhar-tl , yes I've given the new changes a look and the changes suits our needs for now. Thank you for your time and effort, I will proceed to close this discussion for now. :)

**Comment by shridhar-tl** [comment]
@jeffboon97 - If there are no issues in this functionality, please close this discussion and you can start a new discussion for anything not related to this.

In future in case of any feature suggestion or bugs, please use the issues tab instead. Only for questions or other discussions, use discussions tab as there are some technical limitations in using discussions tab for some of my internal use cases with committing code.

Thank you and good day
